src/integrationTest/java/org/opensearch/security/ResourceFocusedTests.java

```diff
@@ -93,9 +87,8 @@ public void testUnauthenticatedFewBig() {
         final String requestPath = "/*/_search";
         final int parrallelism = 5;
         final int totalNumberOfRequests = 100;
-        final boolean statsPrinter = false;
 
-        runResourceTest(size, requestPath, parrallelism, totalNumberOfRequests, statsPrinter);
+        runResourceTest(size, requestPath, parrallelism, totalNumberOfRequests);
```

Comment:
Nice - on my local machine it was more than beefy enough to finish these tests with good passing rate.  Previous the pass/fail was _ignored_, need to tweak these numbers for headroom on the GHA instance to be reliable.

```
Suite: Test class org.opensearch.security.ResourceFocusedTests
  2> ^^^Original size was 218,000,014 bytes, compressed to 845,762 bytes, ratio 257.76
  2> ^^^Original size was 232 bytes, compressed to 95 bytes, ratio 2.44
  2> java.lang.RuntimeException: Unable to wait for all futures to compete, 4064 have finished.
        at __randomizedtesting.SeedInfo.seed([F42A35B8FA7CC14D:7CC120F2EEA22BBC]:0)
        at org.opensearch.test.framework.AsyncActions.getAll(AsyncActions.java:53)
        at org.opensearch.security.ResourceFocusedTests.runResourceTest(ResourceFocusedTests.java:130)
        at org.opensearch.security.ResourceFocusedTests.testUnauthenticatedTonsSmall(ResourceFocusedTests.java:113)

        Caused by:
        java.util.concurrent.TimeoutException
            at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
            at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
            at org.opensearch.test.framework.AsyncActions.getAll(AsyncActions.java:50)
            ... 2 more
  2> ^^^Original size was 218,014 bytes, compressed to 956 bytes, ratio 228.05
  2> java.lang.RuntimeException: Unable to wait for all futures to compete, 808 have finished.
        at __randomizedtesting.SeedInfo.seed([F42A35B8FA7CC14D:7FEB8586C3A28816]:0)
        at org.opensearch.test.framework.AsyncActions.getAll(AsyncActions.java:53)
        at org.opensearch.security.ResourceFocusedTests.runResourceTest(ResourceFocusedTests.java:130)
        at org.opensearch.security.ResourceFocusedTests.testUnauthenticatedManyMedium(ResourceFocusedTests.java:102)

        Caused by:
        java.util.concurrent.TimeoutException
            at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
            at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
            at org.opensearch.test.framework.AsyncActions.getAll(AsyncActions.java:50)
            ... 2 more
            
```

Reply:
This looks very good

Reply:
Just pushed tweaks in 67d7a58 to modify numbers based off what I was seeing on my local host, we will see what the GitHub runners returns when it runs

```
testUnauthenticatedManyMedium-INFO  AsyncActions:53 - Starting to wait for 10000 futures to complete in for 30 seconds.
testUnauthenticatedManyMedium-INFO  AsyncActions:66 - Waited for 17 seconds, completion speed was on average 1.74ms per future 573.59x per second.
testUnauthenticatedTonsSmall-INFO  AsyncActions:53 - Starting to wait for 15000 futures to complete in for 30 seconds.
testUnauthenticatedTonsSmall-INFO  AsyncActions:66 - Waited for 13 seconds, completion speed was on average 0.93ms per future 1073.11x per second.
testUnauthenticatedFewBig-INFO  AsyncActions:53 - Starting to wait for 100 futures to complete in for 30 seconds.
testUnauthenticatedFewBig-INFO  AsyncActions:66 - Waited for 0 seconds, completion speed was on average 2.36ms per future 423.73x per second.
```
_This message was formatted to fit on the screen_

Reply:
Going to dial UP the time allowed to give this a better chance results in 3 minutes is almost silly fast - getting them in 10 should be fine.
```
  2> java.lang.RuntimeException: Unable to wait for all futures to compete, of 15000 futures 6352 have finished.Waited for 30 seconds, completion speed was on average 4.72ms per future 211.64x per second.
        at __randomizedtesting.SeedInfo.seed([AD4FF687B287D014:25A4E3CDA6593AE5]:0)
        at org.opensearch.test.framework.AsyncActions.getAll(AsyncActions.java:62)
        at org.opensearch.security.ResourceFocusedTests.runResourceTest(ResourceFocusedTests.java:130)
        at org.opensearch.security.ResourceFocusedTests.testUnauthenticatedTonsSmall(ResourceFocusedTests.java:113)
  2> ^^^Original size was 218,000,014 bytes, compressed to 845,762 bytes, ratio 257.76
  2> ^^^Original size was 218,014 bytes, compressed to 956 bytes, ratio 228.05
  2> java.lang.RuntimeException: Unable to wait for all futures to compete, of 10000 futures 4107 have finished.Waited for 30 seconds, completion speed was on average 7.31ms per future 136.89x per second.
        at __randomizedtesting.SeedInfo.seed([AD4FF687B287D014:268E46B98B59994F]:0)
        at org.opensearch.test.framework.AsyncActions.getAll(AsyncActions.java:62)
        at org.opensearch.security.ResourceFocusedTests.runResourceTest(ResourceFocusedTests.java:130)
        at org.opensearch.security.ResourceFocusedTests.testUnauthenticatedManyMedium(ResourceFocusedTests.java:102)
```

Reply:
They are passing with decent thresholds, seems to gain speed as it goes.

```
2023-10-26 18:21:37 TEST-ResourceFocusedTests.testUnauthenticatedManyMedium-seed#[A4FEE20A65FB07B1] INFO  AsyncActions:52 - Starting to wait for 10000 futures to complete in for 120 seconds.
2023-10-26 18:22:14 TEST-ResourceFocusedTests.testUnauthenticatedManyMedium-seed#[A4FEE20A65FB07B1] INFO  AsyncActions:71 - Waited for 37 seconds, completion speed was on average 3.70ms per future 270.23x per second.
2023-10-26 18:22:14 TEST-ResourceFocusedTests.testUnauthenticatedTonsSmall-seed#[A4FEE20A65FB07B1] INFO  AsyncActions:180 - Original size was 232 bytes, compressed to 95 bytes, ratio 2.44
2023-10-26 18:22:14 TEST-ResourceFocusedTests.testUnauthenticatedTonsSmall-seed#[A4FEE20A65FB07B1] INFO  AsyncActions:52 - Starting to wait for 15000 futures to complete in for 120 seconds.
2023-10-26 18:22:57 TEST-ResourceFocusedTests.testUnauthenticatedTonsSmall-seed#[A4FEE20A65FB07B1] INFO  AsyncActions:71 - Waited for 42 seconds, completion speed was on average 2.83ms per future 353.39x per second.
2023-10-26 18:22:58 TEST-ResourceFocusedTests.testUnauthenticatedFewBig-seed#[A4FEE20A65FB07B1] INFO  AsyncActions:180 - Original size was 218,000,014 bytes, compressed to 845,762 bytes, ratio 257.76
2023-10-26 18:22:58 TEST-ResourceFocusedTests.testUnauthenticatedFewBig-seed#[A4FEE20A65FB07B1] INFO  AsyncActions:52 - Starting to wait for 100 futures to complete in for 120 seconds.
2023-10-26 18:22:59 TEST-ResourceFocusedTests.testUnauthenticatedFewBig-seed#[A4FEE20A65FB07B1] INFO  AsyncActions:71 - Waited for 0 seconds, completion speed was on average 5.32ms per future 187.97x per second.
```